### ISSUE TITLE
How to set the forecasting sequence length T ′ in your code

### ISSUE THREAD

**Issue by fangzuliang** [gaozhangyang/SimVP-Simpler-yet-Better-Video-Prediction#4]
![image](https://user-images.githubusercontent.com/48270132/180969951-883674b6-d84e-4d36-861d-493fb42db302.png)
![image](https://user-images.githubusercontent.com/48270132/180970398-d167f355-aed2-4c2e-8d8e-388dc48834b0.png)

In your master code, the forecasting sequence length must be same as input sequence length T. but in your article, those two T could be different. Could you please implement the function setting dynamic forecasting sequence length?

**Comment by gaozhangyang** [comment]
The length of the model output and the length of the input are always the same. I think you are concerned with predicting T' output frames and T' $\neq$ T. If T'<T, you choose the top T' output as predictions. Otherwise, you can use the previously predicted frames as input for the next prediction until you get T' predicted frames. For example, if you need to predict 20 frames while the input supports 10 frames, you can input X (t=-9,-8,...,0) to predict Y1 (t=1,2,...,10), then enter Y1 (t=1,2,...,10) to predict Y2 (t=11,12,...,20). Finally, you can concatenate Y1 and Y2 as the prediction Y.

**Comment by LeoXinhaoLee** [comment]
Hi, thank you very much for releasing code for this inspiring work. Regarding the prediction length, say we input 10 previous frames and output 10 future frames, if our final goal is to predict 1 future frame, do we calculate and backpropagate the loss on the other 9 future frames **during training** or just on the first future frame? 

Thank you very much for your time and help!
